packages/firestore/src/local/memory_persistence.ts

```diff
@@ -49,17 +59,31 @@ export class MemoryPersistence implements Persistence {
    */
   private mutationQueues: { [user: string]: MutationQueue } = {};
   private remoteDocumentCache = new MemoryRemoteDocumentCache();
-  private queryCache = new MemoryQueryCache();
+  private readonly queryCache: MemoryQueryCache; // = new MemoryQueryCache();
+  private readonly listenSequence = new ListenSequence(0);
 
   private _started = false;
 
+  private _referenceDelegate: ReferenceDelegate | null;
```

Comment:
We have been using just `ReferenceDelegate` as a the type if a member is essentially always set. Would it be possible to pass the delegate into the constructor from your factory function? You could then also make the variable itself public and remove the getter.